src/VirtualClient/VirtualClient.Contracts/Extensions/MetricExtensions.cs

```diff
@@ -0,0 +1,188 @@
+                }
+            }
+
+            return filteredMetrics;
+        }
+
+        /// <summary>
+        /// Print metrics to console. Default to critical only.
+        /// </summary>
+        /// <param name="metrics">List of Metrics</param>
+        /// <param name="scenario">Scenario</param>
+        /// <param name="criticalOnly">Boolean to note if prints critical only metric</param>
+        public static void PrintConsole(this IEnumerable<Metric> metrics, string scenario, bool criticalOnly = true)
+        {
+            try
+            {
+                Console.WriteLine();
+                Dictionary<string, int> colWidths = new Dictionary<string, int>();
+
+                DataTable table = new DataTable();
+                table.Columns.Add("Scenario", typeof(string));
+                table.Columns.Add("Name", typeof(string));
+                table.Columns.Add("Value", typeof(double));
+                table.Columns.Add("Unit", typeof(string));
+
+                IEnumerable<Metric> metricsToPrint = criticalOnly ? metrics.Where(m => m.Verbosity == MetricVerbosity.Critical).ToList() : metrics;
+
+                foreach (Metric metric in metricsToPrint)
+                {
+                    DataRow row = table.NewRow();
+                    row["Scenario"] = scenario; // You can modify this as needed
+                    row["Name"] = metric.Name;
+                    row["Value"] = metric.Value;
+                    row["Unit"] = metric.Unit;
+                    table.Rows.Add(row);
+                }
+
+                MetricExtensions.PrintDataTableFormatted(table);
+            }
+            catch (Exception exc)
+            {
+                // Don't throw on print console errors
+                Console.WriteLine($"Failed to print metrics on console: '{exc.Message}'");
+            }
+        }
+
+        /// <summary>
+        /// Data table with formated column width.
+        /// </summary>
+        /// <param name="dataTable">Input data table.</param>
+        private static void PrintDataTableFormatted(DataTable dataTable)
+        {
+            Console.WriteLine();
+            Dictionary<string, int> colWidths = new Dictionary<string, int>();
+
+            foreach (DataColumn col in dataTable.Columns)
+            {
+                Console.Write("| " + col.ColumnName);
+                var maxLabelSize = dataTable.Rows.OfType<DataRow>()
+                        .Select(m => (m.Field<object>(col.ColumnName)?.ToString() ?? string.Empty).Length)
+                        .OrderByDescending(m => m).FirstOrDefault();
+
+                maxLabelSize = Math.Max(col.ColumnName.Length, maxLabelSize);
+
+                colWidths.Add(col.ColumnName, maxLabelSize);
+                for (int i = 0; i < maxLabelSize - col.ColumnName.Length + 1; i++)
+                {
+                    Console.Write(" ");
+                }
+            }
+
+            Console.Write("|");
```

Comment:
Use ConsoleLogger.Default vs. Console.

Apply to all

Reply:
Won't change as discussed as console logger only prints line by line